diff -up gnome-settings-daemon-2.21.90.1/plugins/keyboard/gsd-keyboard-manager.c.load-xkb gnome-settings-daemon-2.21.90.1/plugins/keyboard/gsd-keyboard-manager.c
--- gnome-settings-daemon-2.21.90.1/plugins/keyboard/gsd-keyboard-manager.c.load-xkb	2008-02-07 22:49:45.000000000 -0500
+++ gnome-settings-daemon-2.21.90.1/plugins/keyboard/gsd-keyboard-manager.c	2008-02-07 22:49:57.000000000 -0500
@@ -410,6 +410,7 @@ gsd_keyboard_manager_start (GsdKeyboardM
         /* Essential - xkb initialization should happen before */
         gsd_keyboard_xkb_set_post_activation_callback ((PostActivationCallback) gsd_load_modmap_files, NULL);
         gsd_keyboard_xkb_init (client);
+        gsd_keyboard_xkb_load (client);
 
         register_config_callback (manager,
                                   GSD_KEYBOARD_KEY,
