From fffaf9c33100a82a55e8a8b438608347122dba69 Mon Sep 17 00:00:00 2001
From: Gary Ching-Pang Lin <chingpang@gmail.com>
Date: Fri, 14 Oct 2011 15:58:00 +0800
Subject: [PATCH] power: Emit 'Changed' signal to all listeners

---
 plugins/power/gsd-power-manager.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/plugins/power/gsd-power-manager.c b/plugins/power/gsd-power-manager.c
index 4072c02..a5fc50d 100644
--- a/plugins/power/gsd-power-manager.c
+++ b/plugins/power/gsd-power-manager.c
@@ -336,7 +336,7 @@ engine_emit_changed (GsdPowerManager *manager)
         if (manager->priv->connection == NULL)
                 return;
         ret = g_dbus_connection_emit_signal (manager->priv->connection,
-                                             GSD_DBUS_SERVICE,
+                                             NULL,
                                              GSD_POWER_DBUS_PATH,
                                              GSD_POWER_DBUS_INTERFACE,
                                              "Changed",
-- 
1.7.3.4

